app/controllers/katello/api/v2/activation_keys_controller.rb

```diff
@@ -330,5 +324,22 @@ def activation_key_params
 
       key_params
     end
+
+    def parent_scope
+      {}
+    end
+
+    def action_permission
```

Comment:
Apparently creating permissions via permission_generator.rb is not enough to map controller actions to permissions. Those are only for route level authorization. These permissions actually check the filters for the entity found by find_resource.